### PR TITLE
doc(readme): adding TOC and refactoring the content

### PR DESCRIPTION
**Pull request by mynktl**
Changes:
- Added TOC for the readme
- Changed the sequence of section according to usability

Signed-off-by: mayank <mayank.patel@mayadata.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
 To do backup/restore of OpenEBS CStor volumes through Velero utility, you need to install and configure
 OpenEBS velero-plugin.
 
+[![Build Status](https://travis-ci.org/openebs/velero-plugin.svg?branch=master)](https://travis-ci.org/openebs/velero-plugin)
+[![Go Report](https://goreportcard.com/badge/github.com/openebs/velero-plugin)](https://goreportcard.com/report/github.com/openebs/velero-plugin)
+
+## Table of Contents
+- [Compatibility matrix](#compatibility-matrix)
+- [Prerequisite for velero-plugin](#prerequisite-for-velero-plugin)
+- [installation of velero-plugin](#installation-of-velero-plugin)
```
Comment (ranjithwingrider) [inline README.md]: Change to "Installation"

File: README.md [hunk 2]
```diff
@@ -21,8 +45,8 @@ Run the following command to install OpenEBS velero-plugin
 
 This command will add an init container to Velero deployment to install the OpenEBS velero-plugin.
```
Comment (ranjithwingrider) [inline README.md]: In this section, first sentence is providing the step to install velero-plugin with ci image. So you can mention like this was "install development image of velero-pugin".

File: README.md [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
+- [Managing Backups](#managing-backups)
+  - [Creating a backup](#creating-a-backup)
+  - [Creating a restore from backup](#creating-a-restore-from-backup)
+  - [Creating a scheduled backup](#creating-a-scheduled-backup-or-incremental-backup-for-cstor-volume)
+  - [Restoring from a scheduled backup](#restoring-from-a-scheduled-backup)
+
+## Compatibility matrix
+
+|     Image                |    Codebase     |  Velero v0.10.0  | Velero v0.11.0 | Velero v1.0.0 | OpenEBS/Maya release |
+|   -------------------    |  ---------------|   ---------      |  --------------- |  -----------------  | -------- |
```
Comment (kmova) [inline README.md]: Just thinking aloud here. The user might start with a given OpenEBS version and then Velero version and ask for what is the correct Velero Pluggin to install. 

- So should the table be of the format:
  column 1 - openebs/maya version
  column 2 - velero version 
  column 3 - velero plugin image
  column 4 - code base.

Comment (mynktl) [reply README.md]: yes. That make sense.

**Pull request merged**